aiohttp/client.py

```diff
@@ -539,7 +542,9 @@ async def _request(
         try:
             with timer:
                 # https://www.rfc-editor.org/rfc/rfc9112.html#name-retrying-requests
-                retry_persistent_connection = method in IDEMPOTENT_METHODS
+                retry_persistent_connection = (
+                    False if self._test else method in IDEMPOTENT_METHODS
```

Comment:
This can probably become:
```suggestion
                    self._retry_connection and method in IDEMPOTENT_METHODS
```